addons/mail/static/tests/owl/component/chat_window_manager_tests.js

```diff
@@ -601,6 +601,423 @@ QUnit.test('chat window: open / close', async function (assert) {
     assert.verifySteps(['rpc:channel_minimize']);
 });
 
+QUnit.test('chat window: state conservation on toggle home menu', async function (assert) {
+    assert.expect(10);
+
+    // HELPERS SETUP & START
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: 'channel',
+                id: 20,
+                is_minimized: false,
+                name: "General",
+                state: 'open',
+                uuid: 'channel-20-uuid',
+            }],
+        },
+    });
+    const channelId = 20;
+    const localChannelId = `mail.channel_${channelId}`;
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'channel_fetch_preview') {
+                return [{
+                    id: 20,
+                    last_message: {
+                        author_id: [7, "Demo"],
+                        body: "<p>test</p>",
+                        channel_ids: [20],
+                        id: 100,
+                        message_type: 'comment',
+                        model: 'mail.channel',
+                        res_id: 20,
+                    },
+                }];
+            } else if (args.method === 'channel_fold') {
+                return;
+            } else if (args.method === 'message_fetch') {
+                return [...Array(20).keys()].map(i => {
+                    return {
+                        author_id: [11, "Demo"],
+                        body: "<p>body</p>",
+                        channel_ids: [20],
+                        date: "2019-04-20 10:00:00",
+                        id: i + 10,
+                        message_type: 'comment',
+                        model: 'mail.channel',
+                        record_name: 'General',
+                        res_id: i + 1,
+                    }
+                });
+            }
+            return this._super(...arguments);
+        },
+    });
+
+    // Open thread
+    document
+        .querySelector(`.o_MessagingMenu_toggler`)
+        .click()
+    ;
+    await testUtils.nextTick(); // re-render
+    document
+        .querySelector(`
+            .o_MessagingMenu_dropdownMenu
+            .o_ThreadPreviewList_preview`)
+        .click();
+    await testUtils.nextTick(); // re-render
+
+    // SETUP INITIAL STATE
+    // - Set a scroll position to chat window
+    document.querySelector(`.o_MessageList.o_Thread_messageList`).scrollTop = 142;
+
+    // - Set html content of the composer of the chat window
+    let composerTextInput = document.querySelector(`
+        .o_ComposerTextInput
+        > .note-editor
+        > .note-editing-area
+        > .note-editable`);
+    composerTextInput.innerHTML = "XDU for the win !";
+    composerTextInput.dispatchEvent(new window.KeyboardEvent('input'));
```

Comment:
what's the purpose of this LOC?

Reply:
When setting innerHtml, summernote is not aware that something has been added, triggering the input keyboardevent makes it act like you actually added something (he moves the caret at the end for example)

Reply:
But with the execCommand change you requested, this is no more useful